controllers/machineset_controller.go

```diff
@@ -183,3 +207,27 @@ func (r *ReconcileMachineSet) updateLabelsInNode(ctx context.Context, machine *m
 
 	return nil
 }
+
+func (r ReconcileMachineSet) updateTaintsInNode(ctx context.Context, machine *machinev1.Machine) error {
+
+	if machine.Status.NodeRef == nil || machine.Status.NodeRef.Name == "" {
+		return nil
+	}
+	node, err := m.GetNodeForMachine(r, machine)
```

Comment:
If you rebase your commit to `main`, you shouldn't need to get the node again, but pass it in from the outside.